src/initialization/MOM_state_initialization.F90

```diff
@@ -1823,15 +1823,17 @@ subroutine set_Flather_Bdry_Conds(OBC, tv, h, G, PF, tracer_Reg)
     ! Determine where u points are applied at east side 
     do j=jsd,jed ; do I=IsdB,IedB
       if ((I+G%idg_offset) == east_boundary) then !eastern side
-        OBC%OBC_mask_u(I,j) = .true.
-        OBC%OBC_kind_u(I,j) = OBC_FLATHER_E
-        if ((i+1>isd) .and. (i+1<ied) .and. (J>JsdB) .and. (J<JedB)) then
-          OBC%OBC_mask_v(i+1,J) = .true.
-          if (OBC%OBC_kind_v(i+1,J) == OBC_NONE) OBC%OBC_kind_v(i+1,J) = OBC_FLATHER_E
-        endif
-        if ((i+1>isd) .and. (i+1<ied) .and. (J-1>JsdB) .and. (J-1<JedB)) then
-          OBC%OBC_mask_v(i+1,J-1) = .true.
-          if (OBC%OBC_kind_v(i+1,J-1) == OBC_NONE) OBC%OBC_kind_v(i+1,J-1) = OBC_FLATHER_E
+        if (G%mask2dCu(I,j) > 0.99) then
```

Comment:
I did the update and push to user/ksh/open_bc.

The loop ranges are perfect as I have them. They needed to be extended to
cover the T-point corners just outside the computational domain, otherwise
it fails in an ALE remapping. So, point 4,4 and similar on all the tiles. I
think it was in MOM_temp_salt_initialize_from_Z.

Cheers,
Kate

On Thu, Jun 9, 2016 at 12:10 PM, Alistair Adcroft <notifications@github.com>
wrote:

> In src/initialization/MOM_state_initialization.F90
> <https://github.com/NOAA-GFDL/MOM6/pull/314#discussion_r66512762>:
>
> > @@ -1823,15 +1823,17 @@ subroutine set_Flather_Bdry_Conds(OBC, tv, h, G, PF, tracer_Reg)
> >      ! Determine where u points are applied at east side
> >      do j=jsd,jed ; do I=IsdB,IedB
> >        if ((I+G%idg_offset) == east_boundary) then !eastern side
> > -        OBC%OBC_mask_u(I,j) = .true.
> > -        OBC%OBC_kind_u(I,j) = OBC_FLATHER_E
> > -        if ((i+1>isd) .and. (i+1<ied) .and. (J>JsdB) .and. (J<JedB)) then
> > -          OBC%OBC_mask_v(i+1,J) = .true.
> > -          if (OBC%OBC_kind_v(i+1,J) == OBC_NONE) OBC%OBC_kind_v(i+1,J) = OBC_FLATHER_E
> > -        endif
> > -        if ((i+1>isd) .and. (i+1<ied) .and. (J-1>JsdB) .and. (J-1<JedB)) then
> > -          OBC%OBC_mask_v(i+1,J-1) = .true.
> > -          if (OBC%OBC_kind_v(i+1,J-1) == OBC_NONE) OBC%OBC_kind_v(i+1,J-1) = OBC_FLATHER_E
> > +        if (G%mask2dCu(I,j) > 0.99) then
>
> Oh! And this is exactly the generalization we were talking about needing,
> so thanks for this.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-GFDL/MOM6/pull/314/files/5c7991446fffc46729dc7a16a523ade9c42f71ad#r66512762>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAbIHuL8WIOfOusGHxxYnmFy_F8WrUGxks5qKHNTgaJpZM4Ixds->
> .
>